docs/messaging/messaging-spans.md

```diff
@@ -512,6 +512,50 @@ flowchart LR;
 | `messaging.message.id` | `"a1"` | `"a2"` | |
 | `messaging.batch.message_count` |  |  | 2 |
 
+### Batch publishing
+
+Given is a publisher that publishes a batch with two messages to a topic "Q" on
+Kafka, and two different consumers receiving one of the messages.
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  direction TB
+  CA[Span Create A]
+  CB[Span Create B]
+  P[Span Publish]
+  end
+  subgraph CONSUMER1
+  direction TB
+  D1[Span Receive A]
+  end
+  subgraph CONSUMER2
+  direction TB
+  D2[Span Receive B]
+  end
+  CA-. link .-P;
+  CB-. link .-P;
+  CA-. link .-D1;
+  CB-. link .-D2;
+
+  classDef normal fill:green
+  class P,CA,CB,D1,D2 normal
+  linkStyle 0,1,2,3 color:green,stroke:green
+```
```

Comment:
I added some colors to the diagram so it's easier to distinguish the links

```suggestion
```mermaid
flowchart TD;
  subgraph CONSUMER2
  direction LR
  RB[Receive B]
  end
  subgraph CONSUMER1
  direction LR
  RA[Receive A]
  end
  subgraph PRODUCER
  direction LR
  CA[Create A]
  CB[Create B]
  P[Publish]
  end
  %% Link 0
  CA-. link .-P;  
  %% Link 1
  CB-. link .-P;
  %% Link 2
  CA-. link .-RA;
  %% Link 3
  CB-. link .-RB;

  %% Style the nodes and corresponding links
  %% producer links and nodes
  classDef producer fill:green
  class P,CA,CB producer
  linkStyle 0,1 color:green,stroke:green

  %% consumer1 links and nodes with light blue
  classDef consumer1 fill:#9eecff,color:black
  class RA consumer1
  linkStyle 2 color:#9eecff,stroke:#9eecff

  %% consumer2 links and nodes with dark blue
  classDef consumer2 fill:#032a61
  class RB consumer2
  linkStyle 3 color:#032a61,stroke:#032a61
```
```

Reply:
I'd prefer using too many colors, as it might confuse readers.

To make the meaning of colors clear, I added [this paragraph from OTEP 220](https://github.com/open-telemetry/oteps/blob/main/text/trace/0220-messaging-semantic-conventions-span-structure.md#examples) to the beginning of the "Examples" section:

> This section contains a list of examples illustrating the use of the conventions outlined above. Green boxes denote spans that are required to exist in order to conform to those conventions. Other boxes denote spans that are not required and covered by the conventions, but are hopefully helpful in understanding how messaging spans can be integrated into an overall trace flow. Solid arrows denote parent/child relationships, dotted arrows denote link relationships.